evap/settings.py

```diff
@@ -402,6 +404,14 @@ def CHARACTER_ALLOWED_IN_NAME(character):  # pylint: disable=invalid-name
 OIDC_OP_JWKS_ENDPOINT = "https://example.com/certs"
 
 
+### Content Security Policy
+CSP_OBJECT_SRC = ("'none'", )
+CSP_SCRIPT_SRC = ("'strict-dynamic'", )  # scripts require a correct nonce or to be loaded by a trusted script
+CSP_STYLE_SRC = ("'unsafe-inline'", "'self'") # inline-styles are allowed
+CSP_IMG_SRC = ("'self'", "data:") # images from same host or as data: url are allowed
+CSP_BASE_URI = ("'none'", )
+CSP_INCLUDE_NONCE_IN=['script-src']
```

Comment:
The current header looks something like (linebreaks inserted by me)
```plain
default-src 'self';
script-src 'strict-dynamic' 'nonce-stdrWony6OaQQhaUMqHNCQ==';
img-src 'self' data:;
object-src 'none';
base-uri 'none';
style-src 'unsafe-inline' 'self'
```
In contrast to the google article linked at some point, this includes a `default-src` which as far as I understand is the fallback for any unspecified sources. The examples of "strict CSP" on the internet omit a `default-src`. There are some mentions in the spec though that suggest that the most secure configuration would be `default-src: 'none'` (https://www.w3.org/TR/2024/WD-CSP3-20240115/#example-ae46ad12 , https://www.w3.org/TR/2024/WD-CSP3-20240115/#example-ae46ad12). What do you think?

Reply:
You linked the same example ([Example 26](https://www.w3.org/TR/2024/WD-CSP3-20240115/#example-ae46ad12) from 8.6. Exfiltration) twice. The point in the documentation here is about exfiltrating data. An attack scenario requires that an attacker makes the browser send a request to a server that the attacker controls, or at least can observe traffic at. From my understanding, both `default-src: 'none'` and `default-src: 'self'` prevent this exfiltration scenario: With `'self'`, we only allow requests to the same origin as the current website (-> evap.hpi.de), and attackers can not monitor HTTPS encrypted traffic to our server. If they could, they'd have much better attack vectors than XSS.

I think `'self'` is a sane default, for most kinds of resources (images, CSS, javascript, connect-src, media, prefetch, fonts). Restricting it to `'none'` seems to be a way to make us struggle with CSP-caused errors forever.

Reply:
Whoops, updated.

Agree that `'self'` is sane, I was just wondering if there was anything in particular that caused the divergence from the article. Example 8 "recommends" to start with a default source of none and build up a policy from that (and I think it would be interesting to see if there is anything we missed here), but I agree that would probably be more annoyed by changing browser behavior or so. In any case, merging this PR has a higher priority to me than figuring out the ultimate ruleset

Reply:
Agreed to default to 'self' for now in an offline discussion. It prevents exfiltration attacks and the average user shouldn't be able to make our webserver serve their files, so it mostly also restricts the content to trusted files

(Explicit consideration: Grade documents in combination with an XSS vulnerability could be used to inject attacker-controlled images/css/whatver, but we're fine with that for now. Javascript would still require the nonce)